app/src/main/java/org/groundplatform/android/ui/datacollection/tasks/photo/PhotoTaskFragment.kt

```diff
@@ -61,25 +61,22 @@ class PhotoTaskFragment : AbstractTaskFragment<PhotoTaskViewModel>() {
   lateinit var homeScreenViewModel: HomeScreenViewModel
 
   // Registers a callback to execute after a user captures a photo from the on-device camera.
-  private var capturePhotoLauncher: ActivityResultLauncher<Uri> =
-    registerForActivityResult(ActivityResultContracts.TakePicture()) { result: Boolean ->
-      externalScope.launch {
-        if (result) {
-          if (isViewModelInitialized) {
-            viewModel.savePhotoTaskData(capturedPhotoUri)
-          } else {
-            pendingCapturedPhotoUri = capturedPhotoUri
-            pendingCaptureTimestamp = System.currentTimeMillis()
-          }
-        }
-      }
-    }
+  private lateinit var capturePhotoLauncher: ActivityResultLauncher<Uri>
 
   private var hasRequestedPermissionsOnResume = false
   private var taskWaitingForPhoto: String? = null
   private var capturedPhotoPath: String? = null
   private lateinit var capturedPhotoUri: Uri
 
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+
+    capturePhotoLauncher =
+      registerForActivityResult(ActivityResultContracts.TakePicture()) { result: Boolean ->
+        externalScope.launch(Dispatchers.IO) { handleCaptureResult(result) }
```

Comment:
Can we inject `IoDispatcher` instead of using `Dispatchers.IO`? This would allow testing the feature using test dispatchers in unit tests.